span_test.go

```diff
@@ -363,7 +363,6 @@ func TestSpan_References(t *testing.T) {
 }
 
 func TestSpanContextRaces(t *testing.T) {
-	t.Skip("Skipped: test will panic with -race, see https://github.com/jaegertracing/jaeger-client-go/issues/526")
```

Comment:
It might be useful to add a few more mutators to this test, namely those that affect samplingState and flags.

Reply:
Yeah added a few, and also called span.Finish in the end, and found a few more data race conditions in the reportSpan() method and modified it